### ISSUE TITLE
Substrate interface

### ISSUE THREAD

**Issue by adlrocha** [consensus-shipyard/lotus#38]
Depends on #33, #36
## Background
#33 will include multi-subnet peer support into Eudico, which means that the same eudico daemon will be able to spawn and handle more than blockchchain instances for IPC (i.e. more than one subnet stack). However, developers may choose to deploy independent instances of subnets in different daemons, and handle their own deployment.

This interface is used by IPC controllers to track the health of the blockchain instances for the subnets is managing and (optionally, and this is a "nice-to-have") spawning, killing, starting and stopping blockchain instances. Thus, when a new subnet is spawned by the IPC client, the subnet controller may trigger the automatic spawn of the subnet (if support as for the case of the multi-peer eudico daemon),  or it will urge the user to deploy the blockchain instance and point to the right endpoint the IPC controller can use to interact with it. The substrate interface will use that endpoint to interact with the blockchain instances of the subnets and track the health of the deployment. 

_(note: alternative implementations of the substrate interface may automate the deployment of blockchain instances using docker, or any other external deployment system for the infrastructure). 

## Goal 
Couple the deployment of blockchain instances (infrastructure) with the IPC client through the substrate interface.

## Tasks.
- [ ] Design the functionality and methods of the substrate interface.
- [ ] Implementation of the substrate interface for health check of blockchain instances and the optional (and automatic) management of their lifecycle.
  -  The blockchain instance interface should include an optional set of methods to spawn/kill/start/stop new subnets if they support the automatic spawn of blockchain stacks (as is the case for our Lotus-based blockchain instances with Eudico (as of #33 Eudico will expose an API to handle the lifecycle of subnet within the same daemon).

**Comment by adlrocha** [comment]
This functionality will become part of the https://github.com/consensus-shipyard/ipc-agent. Closing it here and tracking it on that repo.
